pkg/local-storage/member/csi/node.go

```diff
@@ -106,18 +106,33 @@ func (p *plugin) NodePublishVolume(ctx context.Context, req *csi.NodePublishVolu
 		return resp, nil
 	}
 
-	if req.GetVolumeCapability().GetBlock() != nil {
+	// mount volume
+	volumeCap := req.GetVolumeCapability()
+	switch volumeCap.GetAccessType().(type) {
+	case *csi.VolumeCapability_Block:
 		// raw block
-		return resp, p.mounter.MountRawBlock(devicePath, req.TargetPath)
-	}
-
-	if mnt := req.GetVolumeCapability().GetMount(); mnt != nil {
+		err := p.mounter.MountRawBlock(devicePath, req.TargetPath)
+		if err != nil {
+			return resp, err
+		}
+	case *csi.VolumeCapability_Mount:
 		// filesystem block
-		return resp, p.mounter.FormatAndMount(devicePath, req.TargetPath, mnt.FsType, mnt.MountFlags)
+		mnt := req.GetVolumeCapability().GetMount()
+		err := p.mounter.FormatAndMount(devicePath, req.TargetPath, mnt.FsType, mnt.MountFlags)
+		if err != nil {
+			return resp, err
+		}
+	default:
+		return resp, fmt.Errorf("invalid access type")
 	}
 
-	return resp, fmt.Errorf("invalid access type")
+	// configure volume qos
+	if err := p.configureQoS(ctx, req); err != nil {
```

Comment:
What will happen if fail at configuring QoS, is it can be reconfigured again?

Reply:
@SSmallMonster I'm rewriting the logic this week. 

It is divided into two parts, one is that hwameistor itself supports QoS,, and the other is based on the previous one, which supports configuring volume QoS, through CSI.

For the second point, it is necessary to wait for the relevant kep to be determined before implementing it. Users can still configure QoS by modifying hwameistor's own cr

Reply:
I see, my concern is if `NodePublishVolume` return error at `configureQoS`, is the QoS policy can be applied again? 

Since at the begin of `NodePublishVolume`, it returns directly when found volume has already mounted.

Reply:
> I see, my concern is if NodePublishVolume return error at configureQoS, is the QoS policy can be applied again?

No. This logic will be changed later and I don't want to configure Qos in `NodePublishVolume`.